packages/react/src/use-render/useRender.ts

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import type { ComponentRenderFn } from '../utils/types';
+import { useComponentRenderer } from '../utils/useComponentRenderer';
+import { defaultRenderFunctions } from '../utils/defaultRenderFunctions';
+import { CustomStyleHookMapping as StateAttributes } from '../utils/getStyleHookProps';
+
+/**
+ * Returns a function that renders a Base UI component.
+ */
+function useRender<RenderedElementType extends Element>(
+  settings: useRender.Settings<RenderedElementType>,
+) {
+  const { render, props } = settings;
+  const { ref, ...extraProps } = props ?? {};
+
+  return useComponentRenderer({
+    render,
+    state: {} as Record<string, any>,
```

Comment:
off-topic: I wonder if `state` should be optional in `useComponentRenderer`

Reply:
We probably didn't have use-case for it so far, as all components have some state, but we could consider it if we hit this need again.

Reply:
A lot of them actually define an empty object outside the component (in fact, this needs to be done here to prevent a new reference every render). It would be good to make it optional since many components have empty state objects